src/interface.jl

```diff
@@ -0,0 +1,664 @@
+#######################################
+
+abstract type ADBackend end
+struct ForwardDiffAD <: ADBackend end
+struct TrackerAD <: ADBackend end
+
+const ADBACKEND = Ref(:forward_diff)
+function setadbackend(backend_sym)
+    @assert backend_sym == :forward_diff || backend_sym == :reverse_diff
+    backend_sym == :forward_diff
+    ADBACKEND[] = backend_sym
+end
+
+ADBackend() = ADBackend(ADBACKEND[])
+ADBackend(T::Symbol) = ADBackend(Val(T))
+function ADBackend(::Val{T}) where {T}
+    if T === :forward_diff
+        return ForwardDiffAD
+    else
+        return TrackerAD
+    end
+end
+
+######################
+# Bijector interface #
+######################
+
+"Abstract type for a `Bijector`."
+abstract type Bijector end
+
+Broadcast.broadcastable(b::Bijector) = Ref(b)
+
+"""
+Abstract type for a `Bijector` making use of auto-differentation (AD) to
+implement `jacobian` and, by impliciation, `logabsdetjac`.
+"""
+abstract type ADBijector{AD} <: Bijector end
+
+"""
+    inv(b::Bijector)
+    Inversed(b::Bijector)
+
+A `Bijector` representing the inverse transform of `b`.
+"""
+struct Inversed{B <: Bijector} <: Bijector
+    orig::B
+end
+
+inv(b::Bijector) = Inversed(b)
+inv(ib::Inversed{<:Bijector}) = ib.orig
+
+"""
+    logabsdetjac(b::Bijector, x)
+    logabsdetjac(ib::Inversed{<: Bijector}, y)
+
+Computes the log(abs(det(J(x)))) where J is the jacobian of the transform.
+Similarily for the inverse-transform.
+
+Default implementation for `Inversed{<: Bijector}` is implemented as
+`- logabsdetjac` of original `Bijector`.
+"""
+logabsdetjac(ib::Inversed{<: Bijector}, y) = - logabsdetjac(ib.orig, ib(y))
+
+"""
+    forward(b::Bijector, x)
+    forward(b::Bijector, x, logjac)
+
+Computes both `transform` and `logabsdetjac` in one forward pass, and
+returns a named tuple `(rv=b(x), logabsdetjac=logabsdetjac(b, x))`.
+
+`forward(b::Bijector, x, logjac)` allows the user to specify the accumulation
+variable for the `logabsdetjac` field. This is useful for doing batch computations.
+See below for example.
+
+This defaults to the call above, but often one can re-use computation
+in the computation of the forward pass and the computation of the
+`logabsdetjac`. `forward` allows the user to take advantange of such
+efficiencies, if they exist.
+
+# Examples
+`forward(b::Bijector, x, logjac)` allows specification of the accumulation variable
+for the `logabsdetjac` field. This is useful for doing **batch computations**.
+```
+julia> b = PlanarLayer(2);
+
+julia> cb = b ∘ b;
+
+julia> x = randn(2, 3)
+2×3 Array{Float64,2}:
+  0.0660476  -0.77195  -1.7832  
+ -0.147743   -1.46459   0.264924
+
+julia> forward(cb, x)
+ERROR: MethodError: no method matching +(::Array{Float64,1}, ::Float64)
+  ...
+julia> forward(cb, x, zeros(size(x, 2)))
+(rv = [1.10887 0.32029 -0.704563; -0.639206 -1.97935 -0.243419], logabsdetjac = [0.018534, 1.46352e-5, 0.00521633])
+```
+    
+"""
+forward(b::Bijector, x) = forward(b, x, zero(eltype(x)))
```

Comment:
@torfjelde This is the root cause of the error. Please read the next comments for context.

Reply:
This is why we provided the `forward(b::Bijector, x, logjac)` method, _but_ I've now changed it to what you suggested as I agree it's a much better approach!:)